### PR TITLE
[docs] Revisit the docs pages on clearing bundler caches + RN version mismatches

### PR DESCRIPTION
**Pull request by ide**
Why
---
The information on the pages to clear the bundler caches was outdated or incorrect in some places. I also did an editorial pass over the writing and formatting to explain some of the motivation behind clearing the caches and to make the commands easier to read.

How
---
For the cache-clearing pages, I researched which files and directories are created by Expo CLI and Metro today. Both `expo start` and `react-native init X; cd X; yarn start` produced the same cache files: `haste-map-XXX` and a directory named just `metro-cache` both under the OS's temp directory.

Additionally, I read through parts of the metro-config and metro-cache source code to be more confident that `metro-cache` is the directory that Metro creates and that it's unlikely there are other cache directories. I also read the parts of the expo-cli and xdl source related to the `--clear` option to understand and explain its behavior with a little more detail (it also tells Webpack to clear its caches).

Other changes:

* Specified "bundler caches" in the titles to be more precise about what kinds of caches we're clearing (we're not clearing any device caches).
* I changed the one-line commands to multi-line blocks that are easier to read and still copy-pasteable.
* `expo start -c` -> `expo start --clear` (in scripts, docs, and other places outside of a terminal, writing out the long-form option helps communicate what's happening better than the shorthand options do)
* Technical corrections (e.g., `watchman watch-del-all` does not clear dependencies, `rm -rf del` on Windows should be just `del`)
* OSX -> macOS
* NPM -> npm
* "bare workflow app" -> "bare workflow project" on the version mismatch page (we say "project" everywhere else on that page)

Test Plan
---------
Visited http://localhost:3002/faq/clear-cache-macos-windows/, http://localhost:3002/faq/clear-cache-macos-linux/, and http://localhost:3002/faq/react-native-version-mismatch/.

Verified they all render as expected.

Screen Shot 2021-03-03 at 6 45 09 PM

### CODE REVIEW

File: docs/pages/faq/react-native-version-mismatch.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ The packager that you're running in your terminal (either with `expo start` or `
 - If this is a bare workflow project, and this error is occurring right after upgrading your React Native version, you should double-check that you've performed each of the upgrade steps correctly.
 
 - Finally:
-  - Clear your cache by running `rm -rf node_modules && yarn cache clean && yarn && watchman watch-del-all && rm -rf $TMPDIR/react-native-packager-cache-* && rm -rf $TMPDIR/metro-bundler-cache-* && expo start -c`
-     - Commands if you are using NPM can be found [here.](clear-cache-osx-linux)
+  - Clear your bundler caches by running `rm -rf node_modules && yarn cache clean && yarn && watchman watch-del-all && rm -rf $TMPDIR/haste-map-* && rm -rf $TMPDIR/metro-cache && expo start --clear`
```
Comment (brentvatne) [inline docs/pages/faq/react-native-version-mismatch.md:31]: should probably default to npm, people using yarn will figure out to change it to yarn if needed whereas people who use npm might not even know what yarn is

File: docs/pages/faq/clear-cache-macos-linux.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+title: Clearing bundler caches on macOS and Linux
+---
+
+> Need to clear development caches on Windows? [Find the relevant commands here.](clear-cache-windows.md)
+
+There are a number of different caches associated with your project that can prevent your project from running as intended. Clearing a cache sometimes can help you work around issues related to stale or corrupt data and is often useful when troubleshooting and debugging.
+
+To clear the various caches, run:
```
Comment (brentvatne) [inline docs/pages/faq/clear-cache-macos-linux.md:9]: we should just make a cli tool that does all of these things

**Review by brentvatne — APPROVED** [review]

File: docs/pages/faq/clear-cache-macos-linux.md (hunk 2 above)
Comment (ide) [reply docs/pages/faq/clear-cache-macos-linux.md:9]: lol, `expo forget`

**Review by tcdavis — APPROVED** [review]
Looks great, thanks for digging in more.
Not sure if you want to bring this language over from the other PR:

It is a good habit to understand commands you find on the internet before you run them.  We explain each command below for Expo CLI and Yarn, but the corresponding commands for npm and React Native CLI have the same behavior.

**Comment by ide** [comment]
Go ahead and land your PR and I'll rebase on top of it.

**Pull request merged**